File: src/engine/core/Application.java
package engine.core;

import engine.core.events.*;

public class Application
{
	private static boolean isRunning;
	
	public static void start()
	{
		EventBus.get(ApplicationStartEvent.class).postEvent(new ApplicationStartEvent());
		
		isRunning = true;
		
		while (isRunning)
		{
			EventBus.get(PreUpdateEvent.class).postEvent(new PreUpdateEvent());
			EventBus.get(EarlyUpdateEvent.class).postEvent(new EarlyUpdateEvent());
			EventBus.get(UpdateEvent.class).postEvent(new UpdateEvent());
			EventBus.get(LateUpdateEvent.class).postEvent(new LateUpdateEvent());
			EventBus.get(PostUpdateEvent.class).postEvent(new PostUpdateEvent());
		}
		
		EventBus.get(ApplicationExitEvent.class).postEvent(new ApplicationExitEvent());
	}
	
	public static void stop()
	{
		isRunning = false;
	}
}

File: TakeoverGame/TakeoverGame/src/engine/core/Application.cpp
#include "Application.h"
#include "EventBus.hpp"

using namespace vlk;

namespace
{
	Boolean APPLICATION_RUNNING;
}

void Application::Run()
{
	LogInfo("Application", "Application starting.");
	APPLICATION_RUNNING = true;

	EventBus<ApplicationStartEvent>::Get().PostEvent({});

	while (APPLICATION_RUNNING)
	{
		EventBus<PreUpdateEvent>::Get().PostEvent({});
		EventBus<EarlyUpdateEvent>::Get().PostEvent({});
		EventBus<UpdateEvent>::Get().PostEvent({});
		EventBus<LateUpdateEvent>::Get().PostEvent({});
		EventBus<PostUpdateEvent>::Get().PostEvent({});
	}

	EventBus<ApplicationExitEvent>::Get().PostEvent({});
}

void Application::Exit()
{
	LogInfo("Application", "Application exiting.");
	APPLICATION_RUNNING = false;
}
